bokeh/document.py

```diff
@@ -320,7 +320,7 @@ def add_root(self, model):
             return
         self._push_all_models_freeze()
         try:
-            self._roots.add(model)
+            self._roots.append(model)
```

Comment:
The original code made sense with this choice of vocabulary. After all, models (classes) are unique and order doesn't matter.